core/full_menu_view.js

```diff
@@ -0,0 +1,522 @@
+/* jslint node: true */
+'use strict';
+
+//  ENiGMA½
+const MenuView = require('./menu_view.js').MenuView;
+const ansi = require('./ansi_term.js');
+const strUtil = require('./string_util.js');
+const formatString = require('./string_format');
+const pipeToAnsi = require('./color_codes.js').pipeToAnsi;
+
+//  deps
+const util = require('util');
+const _ = require('lodash');
+
+exports.FullMenuView = FullMenuView;
+
+function FullMenuView(options) {
+  options.cursor = options.cursor || 'hide';
+  options.justify = options.justify || 'left';
+
+
+  MenuView.call(this, options);
+
+
+  // Initialize paging
+  this.pages = [];
+  this.currentPage = 0;
+
+  this.initDefaultWidth();
+
+
+
+  const self = this;
+
+  //  we want page up/page down by default
+  if (!_.isObject(options.specialKeyMap)) {
+    Object.assign(this.specialKeyMap, {
+      'page up': ['page up'],
+      'page down': ['page down'],
+    });
+  }
+
+  this.autoAdjustHeightIfEnabled = function() {
+    if (this.autoAdjustHeight) {
+      this.dimens.height = (this.items.length * (this.itemSpacing + 1)) - (this.itemSpacing);
+      this.dimens.height = Math.min(this.dimens.height, this.client.term.termHeight - this.position.row);
+    }
+
+    this.positionCacheExpired = true;
+  };
+
+  this.autoAdjustHeightIfEnabled();
+
+
+
+  this.getSpacer = function() {
+    return new Array(self.itemHorizSpacing + 1).join(this.fillChar);
+  }
+
+
+  this.clearPage = function() {
+    var width = this.dimens.width;
```

Comment:
Please try to use `let` and `const`, no `var` -- it's very rarely needed.

Reply:
Updated. How embarrassing :) It's been years since I've done a lot of serious JS work and old habits die hard. lol

Reply:
Totally get it. It's hard when you bounce around in different runtimes too